src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,12 +1048,25 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static ReadOnlyCollection<SyntaxKind> _lazyContextualKeywordKindsArray;
```

Comment:
Another way to put this is: don't prematurely optimize.  If this isn't a critical hot path, and you don't have data to measure your change, then there's no need to do this.